perllib/FixMyStreet/App/Form/Waste/Garden/Renew.pm

```diff
@@ -0,0 +1,172 @@
+        my $form = shift;
+        my $c = $form->{c};
+        $c->stash->{per_bin_cost} = $c->cobrand->feature('payment_gateway')->{ggw_cost};
+        return {
+            current_bins => { default => $c->stash->{garden_form_data}->{bins} },
+            name => { default => $c->user->name },
+            email => { default => $c->user->email },
+            phone => { default => $c->user->phone },
+        };
+    },
+    next => 'summary',
+);
+
+has_page summary => (
+    fields => ['tandc', 'submit'],
+    title => 'Renew your green garden waste subscription',
+    template => 'waste/garden/renew_summary.html',
+    update_field_list => sub {
+        my $form = shift;
+        my $c = $form->{c};
+        my $data = $form->saved_data;
+        my $current_bins = $data->{current_bins};
+        my $new_bins = $data->{new_bins};
+
+        my $cost = $c->cobrand->feature('payment_gateway')->{ggw_cost};
+        my $total = ( $new_bins + $current_bins) * $cost;
+
+        my $orig_sub = $c->stash->{orig_sub};
+        if ( $orig_sub ) {
+            $data->{payment_method} = $orig_sub->get_extra_field_value('payment_method');
+            $data->{billing_address} = $orig_sub->get_extra_field_value('billing_address');
+        }
+        $data->{bin_number} = $current_bins + $new_bins;
+        $data->{payment_method} ||= 'credit_card';
+        $data->{billing_address} ||= $c->stash->{property}{address};
+        $data->{display_total} = $total / 100;
+
+        $data->{name} = $c->user->name;
+        $data->{email} = $c->user->email;
+        $data->{phone} = $c->user->phone;
+        return {};
+    },
+    finished => sub {
+        return $_[0]->wizard_finished('process_garden_renew');
+    },
+    next => 'done',
+);
+
+has_page done => (
+    title => 'Subscription renewed',
+    template => 'waste/garden/renewed.html',
+);
+
+has_field current_bins => (
+    type => 'Integer',
+    label => 'Bins already on property',
+    tags => { number => 1 },
+    required => 1,
+    range_start => 1,
+    range_end => 3,
+);
+
+has_field new_bins => (
+    type => 'Integer',
+    label => 'Additional bins required',
+    tags => { number => 1 },
+    required => 1,
+    range_start => 0,
+    range_end => 3,
+);
+
+has_field payment_method => (
+    type => 'Select',
+    label => 'How do you want to pay?',
+    required => 1,
+    widget => 'RadioGroup',
+    options => [
+        { value => 'direct_debit', label => 'Direct Debit', hint => 'Set up your payment details once, and we’ll automatically renew your subscription each year, until you tell us to stop. You can cancel or amend at any time.' },
+        { value => 'credit_card', label => 'Debit or Credit Card' },
+    ],
+);
+
+has_field billing_differ => (
+    type => 'Checkbox',
+    option_label => 'Check if different to collection address',
+    label => "Billing address",
+    tags => {
+        toggle => 'form-billing_address-row'
+    },
+);
+
+has_field billing_address => (
```

Comment:
Can this share code with the garden subscription form, if all these fields are the same?